### DISCUSSION TITLE
Multi Agent case

### DISCUSSION THREAD

**Discussion by ghost** [aidudezzz/deepbots#148] [Q&A]
Is it possible to implement maddpg in deepbots?

**Comment by tsampazk** [comment]
Hello @abzzall, thank you for your interest on _deepbots_! Your question is basically twofold. First, deepbots is compatible with any RL algorithm implementation that is compatible with gym environments. So, by providing your own implementation of an algorithm and using an existing or constructing a new deepbots environment, you are good to go.

Now, MADDPG is an extension of DDPG for multi-agent problems. Unfortunately, deepbots doesn't support multiple agents out-of-the-box, but provides some groundwork to implement your own multi-agent scheme. You can read up more [here](https://deepbots.readthedocs.io/en/latest/how_deepbots_works.html#emitter-receiver-scheme) for the `emitter-receiver scheme` which can be used to implement a multi-agent environment. Such an example that is still in the works can be found on [this PR](https://github.com/aidudezzz/deepworlds/pull/93) and [this branch](https://github.com/stavliv/deepworlds/tree/double-cartpole/examples/double_cartpole) correspondingly.
